### PR TITLE
Various Linux fixes

### PR DESCRIPTION
**Pull request by Mystro256**
Three fixes/commits:

- Use shared gtest library if available, which is appliable for some Linux distros, such as Fedora, see bug report:
https://bugs.dolphin-emu.org/issues/9402

- Fixes to suse/fedora spec file: version bump to 5.0 and polarssl should be mbedtls

- Added manpages for Linux systems, see bug report:
https://bugs.dolphin-emu.org/issues/9403

Thanks!

[Review on Reviewable](https://reviewable.io/reviews/dolphin-emu/dolphin/3705)

### CODE REVIEW

**Comment by ASSympt0te** [comment]
I don't think the version bump is necessary, 5.0 hasn't come out yet

**Comment by Mystro256** [comment]
Considering it's the development branch and 5.0RC has been tagged, I figured it could be safely changed. If not, just cherry pick out that commit. The only other thing I changed in the commit was polarssl -> mbedtls, as the package was renamed in Fedora.

**Comment by aserna3** [comment]
The RC system was scrapped.

File: Installer/dolphin-emu-nogui.6 [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+\fBdolphin-emu-nogui\fR is an emulator for running GameCube, Wii, and Triforce games on
+Windows, Linux, OS X, and recent Android devices.
+.SH SYNOPSIS
+.B dolphin-emu-nogui
+[\fB-h\fR] [\fB-d\fR] [\fB-l\fR] [\fB-e \fIROM\fR] [\fB-b\fR] [\fB-V \fIVIDEO\fR] [\fB-A \fIAUDIO\fR]
+.SH DESCRIPTION
+Dolphin is a Gamecube, Wii and Triforce (the arcade machine based on the
+Gamecube) emulator, which supports full HD video with several enhancements such
+as compatibility with all PC controllers, turbo speed, networked multiplayer,
+and more.
```
Comment (Pringo) [inline Installer/dolphin-emu-nogui.6]: The "C" in GameCube should be capitalized. Also where is this description from? Perhaps the one from the Dolphin homepage should be used instead.

**Comment by Mystro256** [comment]
I've made some changes based on the input provided:
- Reverted back to 4.0.2 in the spec
- Used dolphin's homepage description in the manpage instead

**Comment by BhaaLseN** [comment]
Feel free to squash those last 3 changes into their respective commits to keep things tidy.

**Comment by jcowgill** [comment]
Reviewed 2 of 2 files at r3.
Review status: 2 of 4 files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---

*[Installer/dolphin-emu-nogui.6, line 3 \[r2\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3705#-KC1RZ53EnVVDcjJae4M:-KC1RZ53EnVVDcjJae4N:-1253617000) ([raw file](https://github.com/dolphin-emu/dolphin/blob/db47b390503e34441f009f09db520b0a94da4c68/Installer/dolphin-emu-nogui.6#L3)):*
The NAME section has a particular convention you should follow. See man(7) and man-pages(7).
In effect: the executable should not be bold, and replace "is" with "\-".

---

*[Installer/dolphin-emu-nogui.6, line 7 \[r2\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3705#-KC1St1XntjNKzjWPaGR:-KC1St1XntjNKzjWPaGS:-1484706757) ([raw file](https://github.com/dolphin-emu/dolphin/blob/db47b390503e34441f009f09db520b0a94da4c68/Installer/dolphin-emu-nogui.6#L7)):*
Usually user provided arguments are written in lowercase.

---

*[Installer/dolphin-emu-nogui.6, line 17 \[r2\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3705#-KC1T344-QqgNx4GorMs:-KC1T344-QqgNx4GorMt:65929474) ([raw file](https://github.com/dolphin-emu/dolphin/blob/db47b390503e34441f009f09db520b0a94da4c68/Installer/dolphin-emu-nogui.6#L17)):*
Sort the list of options?

There are also some more options not listed in the readme.
See here: https://github.com/dolphin-emu/dolphin/blob/master/Source/Core/DolphinWX/Main.cpp#L162
And here: https://sources.debian.net/data/main/d/dolphin-emu/5.0~rev8631+dfsg-1/debian/dolphin-emu.6

---

*[Installer/dolphin-emu-nogui.6, line 29 \[r2\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3705#-KC1THUUn_2LnKtqArRi:-KC1THUUn_2LnKtqArRj:1784024248) ([raw file](https://github.com/dolphin-emu/dolphin/blob/db47b390503e34441f009f09db520b0a94da4c68/Installer/dolphin-emu-nogui.6#L29)):*
The commas should not be bold.

---

*[Installer/dolphin-emu-nogui.6, line 39 \[r2\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3705#-KC1V7ttUCGjQM4mlP-W:-KC1V7ttUCGjQM4mlP-X:1052658713) ([raw file](https://github.com/dolphin-emu/dolphin/blob/db47b390503e34441f009f09db520b0a94da4c68/Installer/dolphin-emu-nogui.6#L39)):*
In the man page used in Debian, I've added a FILES section to briefly document the $HOME/.dolphin-emu directory just above here.

---

*[Installer/dolphin-emu.6, line 1 \[r3\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3705#-KC1VTc2W0BdtN7_GEm7:-KC1VTc2W0BdtN7_GEm8:-1167598539) ([raw file](https://github.com/dolphin-emu/dolphin/blob/e6f90b86e3f7536ed1824e5fc3a660fe53230391/Installer/dolphin-emu.6#L1)):*
Everything I've written for dolphin-emu-nogui.6 applies here to this file as well.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3705)*

**Comment by Mystro256** [comment]
Done, squashed into 3 commits.
Also fixed manpage issues as described above, except one issue:

- Sort the list of options: I think just having the options in the order specified in the README and code would be the most consistent choice here. There's no de facto rule when it comes to options sorting, so I don't think it really matters, as it's still legible.

**Comment by AdmiralCurtiss** [comment]
[Google Test recommends against using precompiled system libararies](https://github.com/google/googletest/blob/master/googletest/docs/FAQ.md#why-is-it-not-recommended-to-install-a-pre-compiled-copy-of-google-test-for-example-into-usrlocal), are you sure this is a good idea?

**Comment by jcowgill** [comment]
And then you have other distros (such as Debian / Ubuntu) which ship a shared copy of the source only in /usr/src/gtest to get around the issues Google raised. Should these distros be supported too?

**Comment by delroth** [comment]
In practice there is no reason to care about GTest being statically linked. It's not linked into our main binary, only in unit tests that aren't being distributed to end users.

So I would recommend reverting that part of the pull request and keeping the static linking to our Externals copy.

**Comment by Mystro256** [comment]
I changed shared gtest to be an option that's disabled by default.

I can't make a solution for Debian or Ubuntu, as I don't use these distros and can't test this properly. If desired, an extra check can be added later.

File: CMakeLists.txt [hunk 2]
```diff
@@ -952,6 +960,11 @@ if(${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
 		DESTINATION ${CMAKE_INSTALL_PREFIX}/share/icons/hicolor/48x48/apps)
 	install(FILES Installer/dolphin-emu.desktop
 		DESTINATION ${CMAKE_INSTALL_PREFIX}/share/applications)
+	# Install manpages
+	install(FILES Installer/dolphin-emu.6
+		DESTINATION ${CMAKE_INSTALL_PREFIX}/usr/share/man/man6)
```
Comment (delroth) [inline CMakeLists.txt]: You have an extra /usr there -- CMAKE_INSTALL_PREFIX will usually already be /usr.

File: Installer/dolphin-emu-nogui.6 [hunk 3]
```diff
@@ -0,0 +1,56 @@
+.TH DOLPHIN-EMU-NOGUI 6 "March 4, 2016"
```
Comment (delroth) [inline Installer/dolphin-emu-nogui.6]: I don't believe "Installer" is the right place for this (it was meant for the win32 installer NSIS code, and I'm not sure why people starting putting more stuff in there...). Can you move these files to the Data/ directory?

File: Installer/dolphin-emu.spec [hunk 4]
```diff
@@ -56,7 +56,8 @@ BuildRequires:  lzo-devel
 # Disable miniupnpc in OBS for F20
```
Comment (delroth) [inline Installer/dolphin-emu.spec]: Why are we carrying this file in our repo vs. letting distro maintainers have their own versioning for it? That would make things a lot easier for everyone.

File: CMakeLists.txt (hunk 2 above)
Comment (Mystro256) [inline CMakeLists.txt]: Nice catch, thank you! Typo :)

File: Installer/dolphin-emu-nogui.6 (hunk 3 above)
Comment (Mystro256) [inline Installer/dolphin-emu-nogui.6]: Agreed, I just put it here because the desktop file was here.

Comment (Mystro256) [inline Installer/dolphin-emu-nogui.6]: I can made a commit, where should I stick it? Right in Data? or in a subfolder in Data?

Comment (jcowgill) [inline Installer/dolphin-emu-nogui.6]: I think in right in Data/ is ok (just don't put it in Sys). Once that's done all the Installer/dolphin-emu.* files should be moved as well.

**Comment by Mystro256** [comment]
Alright, I've moved the manpages to Data, and in a separate commit, I moved the Linux specific data files to Data as well.

**Comment by adelpha** [comment]
This adds 5 files to the Windows build. Is that intended?

**Comment by Mystro256** [comment]
No, I'm assuming this is an error in the windows build, where it includes everything from the Data folder (Note that I am not a windows user or developer). The windows build needs to be changed to exclude these files and not just copy the hold Data folder.

**Pull request merged**